src/main/java/com/petfinder/controller/AdvertisementController.java

```diff
@@ -76,4 +80,42 @@ public String latestAdvertisements(@PathVariable int page, Model model) {
         );
         return "adlist";
     }
+
+    @RequestMapping(value = "/newAdd")
+    public String newAdvertisement(@RequestParam(required = false) String title,
+            @RequestParam(required = false) String content,
+            @RequestParam(required = false) String petName,
+            @RequestParam(required = false) Integer age,
+            @RequestParam(required = false) String race,
+            @RequestParam(required = false) String categoryName,
+            @RequestParam(required = false) String voivodership,
+            @RequestParam(required = false) String commune,
+            @RequestParam(required = false) String place,
+            @RequestParam(required = false) String tagsString,
+            @RequestParam(required = false) String attachmentType,
+            @RequestParam(required = false) String attachmentUri,
+            Model model) {
+        if (content != null && petName != null && race != null && categoryName != null && voivodership != null && commune != null && place != null) {
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [Position literals first in String comparisons](https://www.codacy.com/app/matikz1110/PetFinder/file/2538168955/issues/source?bid=3239464&fileBranchId=3285829#l99)